h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -373,9 +384,8 @@ private void gridSearch(Grid<MP> grid) {
         } catch (IllegalArgumentException e) {
           Log.warn("Grid search: construction of model parameters failed! Exception: ", e);
           // Model parameters cannot be constructed for some reason
-          it.modelFailed(model);
-          Object[] rawParams = it.getCurrentRawParameters();
-          grid.appendFailedModelParameters(model != null ? model._key : null, rawParams, e);
+          final Model failedModel = model;
+          it.onModelFailure(failedModel, failedHyperParams -> grid.appendFailedModelParameters(failedModel != null ? failedModel._key : null, failedHyperParams, e));
```

Comment:
This seems unusual to me, imho it should be responsibility of GridSearch to append the failed paramters regardless of what the hyperspacewalker does.

Reply:
it is still GridSearch that is doing the appending, through the lambda/callback.
The contract of `onModelFailure` just say that it should call the callback, without knowing what it does.
Actually, I believe this approach being cleaner and less broken that the previous.

Previously, there were 2 methods, supposedly independent: `getCurrentRawParameters` and `modelFailed`, the first one retrieving some params supposedly stored by the iterator, and the second one handling the failure.
However, the params that we need here are the exact ones used to build the failed model (current impl. failed to see this because of the dissociation between the 2 methods and it is therefore broken when running grid in parallel), so those 2 methods should fundamentally be merged into one, hence the callback.